### PR TITLE
Handle error responses in 'Table.upload_from_file'.

### LINKED ISSUE #2132: bigquery.table.Table.upload_from_file() doesn't check the initial request response status
If you try to add data to a table from a file without a schema:

```python
bigquery_client = bigquery.Client()
dataset = bigquery_client.dataset(dataset_name)
table = dataset.table(table_name)

    with open(source_file_name, 'rb') as source_file:
        job = table.upload_from_file(
            source_file, source_format='text/csv', rewind=True)

job.begin()
```

This will fail with an opaque error:

```
Traceback (most recent call last):
  File "load_data_from_file.py", line 74, in <module>
    args.source_file_anme)
  File "load_data_from_file.py", line 40, in load_data_from_gcs
    source_file, source_format='text/csv', rewind=True)
  File "/Users/jonwayne/workspace/python-docs-samples/bigquery/cloud-client/env/lib/python3.4/site-packages/gcloud/bigquery/table.py", line 913, in upload_from_file
    return client.job_from_resource(json.loads(response_content))
  File "/Users/jonwayne/workspace/python-docs-samples/bigquery/cloud-client/env/lib/python3.4/site-packages/gcloud/bigquery/client.py", line 119, in job_from_resource
    config = resource['configuration']
KeyError: 'configuration'
```

Inspection of the response from the initial upload request reveals the problem:

```
status: 400
body:
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "invalid",
    "message": "Empty schema specified for the load job. Please specify a schema that describes the data being loaded.",
...
```

It seems we should check the response before constructing the job.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #2132.

/cc @jonparrott

### CODE REVIEW

File: google/cloud/bigquery/table.py [hunk 1]
```diff
@@ -776,6 +779,16 @@ def insert_data(self,
 
         return errors
 
+    @staticmethod
```
Comment (dhermes) [inline google/cloud/bigquery/table.py]: Argument for static method vs. just having it be module-private function?

**Comment by dhermes** [comment]
Looks mostly fine other than that nit.

File: google/cloud/bigquery/table.py (hunk 1 above)
Comment (tseaver) [inline google/cloud/bigquery/table.py]: It is closer in the file to the calling method.

Comment (dhermes) [inline google/cloud/bigquery/table.py]: OK.

**Comment by dhermes** [comment]
LGTM though you should make Travis go green first

**Pull request merged**